RELEASE_PROCESS.md

```diff
@@ -39,15 +41,34 @@ Every commit to chef-server master is tested against a full pedant
 run. However, upgrade testing must still be done in advance of the
 release:
 
-- [ ] Run the Chef Delivery-based automated upgrade testing
```

Comment:
These changes are only around upgrade testing, but good feedback. Feel free to push that to this branch!

WRT how much to bump we follow symver http://semver.org/. Part of this process will be reading the commits between last release and now to make sure the changelog is correct, so you should have some idea of what has changed, adding "ask in #chef-server if you are unsure isn't a bad idea.